### ISSUE TITLE
Error in `summary` when no endogenous regressors

### ISSUE THREAD

**Issue by bcallaway11** [john-d-fox/ivreg#9]
Hi --- thanks for the package, this is very useful!  I have a quick question about some functionality when there are no endogenous regressors included in the model.  In particular, in this case (at least for me), the `summary` function errors.  Here is an example:

``` r
library(ivreg)

data("SchoolingReturns", package="ivreg")

iv_res <- ivreg(wage ~ education + experience, 
                ~education + experience, 
                data=SchoolingReturns)
coef(iv_res)
#> (Intercept)   education  experience 
#>  -375.58805    55.05451    25.14156
summary(iv_res)
#> Error in ivdiag(object, vcov. = vcov.): no endogenous/instrument variables
```

I know this is not a main use case; for me, I am using the `ivreg` function internally in another package, and I'd like to be able to cover the case with and without endogenous regressors in the same code.  

For what it's worth, the `AER` package does work in this case:
``` r
data("SchoolingReturns", package="ivreg")

iv_res <- AER::ivreg(wage ~ education + experience, 
                ~education + experience, 
                data=SchoolingReturns)
summary(iv_res)
#> 
#> Call:
#> AER::ivreg(formula = wage ~ education + experience | education + 
#>     experience, data = SchoolingReturns)
#> 
#> Residuals:
#>     Min      1Q  Median      3Q     Max 
#> -647.16 -161.20  -29.57  124.79 1612.62 
#> 
#> Coefficients:
#>             Estimate Std. Error t value Pr(>|t|)    
#> (Intercept) -375.588     37.802  -9.936   <2e-16 ***
#> education     55.055      2.140  25.722   <2e-16 ***
#> experience    25.142      1.383  18.174   <2e-16 ***
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#> 
#> Residual standard error: 238 on 3007 degrees of freedom
#> Multiple R-Squared: 0.1811,  Adjusted R-squared: 0.1805 
#> Wald test: 332.5 on 2 and 3007 DF,  p-value: < 2.2e-16
```

**Comment by john-d-fox** [comment]
I agree that in the case ivreg() should just return the OLS estimates. At least initially, I'll leave this suggestion to Achim Zeleis, who wrote this part of the ivreg code.

Thanks,
 John

**Comment by zeileis** [comment]
The reason is that IV diagnostics cannot be computed in that case but are requested by default. Thus, using

```
summary(iv_res, diagnostics = FALSE)
```

works without problem. The reason that the `AER` implementation behaves differently is just because it has `diagnostics = FALSE` as the default.

I have just pushed a modified version that sets `diagnostics = FALSE` in case there are either no endogenous and/or no instrument variables. Previously, there was already code that did the same if there was no formula part for the instruments. See: https://github.com/john-d-fox/ivreg/commit/467fa5edacb0e5a4a6d9d7a141ff2894271762c8

The previous change of `diagnostics` from the default `TRUE` to `FALSE` was done silently without warning I thought might be potentially confusing. Hence I've changed the default to `NULL` and only set it to `TRUE` if both endogenous and instrument variables are available. And then if diagnostics is actually `TRUE` but no endogenous/instrument variables are available, a warning is issued when changing it to `FALSE`.

```
remotes::install_github("john-d-fox/ivreg")

library("ivreg")
data("SchoolingReturns", package="ivreg")
iv_res <- ivreg(wage ~ education + experience | education + experience,  data = SchoolingReturns)

summary(iv_res)
## Call:
## ivreg(formula = wage ~ education + experience | education + experience, 
##     data = SchoolingReturns)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -647.16 -161.20  -29.57  124.79 1612.62 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(>|t|)    
## (Intercept) -375.588     37.802  -9.936   <2e-16 ***
## education     55.055      2.140  25.722   <2e-16 ***
## experience    25.142      1.383  18.174   <2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 238 on 3007 degrees of freedom
## Multiple R-Squared: 0.1811,	Adjusted R-squared: 0.1805 
## Wald test: 332.5 on 2 and 3007 DF,  p-value: < 2.2e-16 

summary(iv_res, diagnostics = TRUE)
## Call:
## ivreg(formula = wage ~ education + experience | education + experience, 
##     data = SchoolingReturns)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -647.16 -161.20  -29.57  124.79 1612.62 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(>|t|)    
## (Intercept) -375.588     37.802  -9.936   <2e-16 ***
## education     55.055      2.140  25.722   <2e-16 ***
## experience    25.142      1.383  18.174   <2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 238 on 3007 degrees of freedom
## Multiple R-Squared: 0.1811,	Adjusted R-squared: 0.1805 
## Wald test: 332.5 on 2 and 3007 DF,  p-value: < 2.2e-16 
## 
## Warning message:
## In summary.ivreg(iv_res, diagnostics = TRUE) :
##   diagnostics cannot be computed without endogenous/instrument variables
```

**Comment by john-d-fox** [comment]
Thanks for the explanation Achim. Your solution makes sense.

**Comment by zeileis** [comment]
Thanks for the feedback, \documented in the NEWS now (https://github.com/john-d-fox/ivreg/commit/e6bcb0689ee71eaaf4476d1a43f86a50acbae463) and issue closed. OK, @bcallaway11 ?

**Comment by bcallaway11** [comment]
Yes, thank you very much!
